### DISCUSSION TITLE
Dispatch opening wrong CANoe version

### DISCUSSION THREAD

**Discussion by stefanuca** [mhammond/pywin32#2543] [General]
Hello,

So i am trying to open CANoe with dispatch and then start the measurement, the only thing is that i have 2 canoe versions installed, 10 and 11, and i want it to open version 11 but it opens version 10.
i m using the code from the first example i could find, the easiest one

i deleted canoe 10 from all the possible paths in system and account variables and i replaced them with canoe 11 and it still opens canoe 10

the cfg file that i want to open it s for canoe 11.

* Version of Python and pywin32
python version 3.6.2
pywin32 version latest

code:

       from win32com.client import *
       from win32com.client.connect import *
       app = Dispatch('CANoe.Application')  
         # current dir must point to the script file
        cfg = os.path.join(os.curdir, cfgPath)
        cfg = os.path.abspath(cfg)
        print('Opening: ', cfg)
        self.ConfigPath = os.path.dirname(cfg)
        self.Configuration = self.App.Configuration
        self.App.Open(cfg)

Anyone knows anything about this? Thanks

**Comment by sschukat** [comment]
Hello, 
this is not the support forum for PyWin32, so this issue will possibly be closed.

For your problem you have to use the "Version Dependend ProgId" and not the "Version Independent ProgId", since the version independent progid always maps to **one** version dependend progid. In your case I would guess "CANoe.Application" --> "CANoe.Application.10". So either would have to set CANoe 11 as default COM server (support request at Vector) or search for the version dependent progid for CANoe 11 (most likely "CANoe.Application.11")

**Comment by stefanuca** [comment]
@sschukat i will try that, thanks
Ps i tried to put 'CANoe.Application.11' in code and i get an error, wrong parameter

**Comment by mhammond** [comment]
Thanks @sschukat!
